tiledb/api/src/filter/mod.rs

```diff
@@ -8,9 +8,15 @@ use crate::context::{CApiInterface, Context, ContextBound};
 use crate::error::{DatatypeErrorKind, Error};
 use crate::{Datatype, Result as TileDBResult};
 
-pub use crate::filter::list::{Builder as FilterListBuilder, FilterList};
-
+pub mod ftype;
 pub mod list;
+pub mod option;
```

Comment:
`option` should not be a public module, there's no API which would enable users to do anything with it, and I think that's a good thing, as the current `FilterData` is structured in a way to set a filter option if and only if a filter of that type needs it.

I also had a strong initial reaction to seeing `ftype` as a `pub mod`, but I see the case for it being not as fuzzy.  `FilterData` is still how you actually interact with libtiledb, but maybe you just want to know what the thing is without passing data around.  Although I'm not really sure why you would want to do that, unless you were, say, writing tests that the interactions with the C API are doing the thing you think they should be doing.  Seems to me like it would bloat the generated rust docs and create confusion when users start looking at how to create the filters, since you'd be offering them a whole enum that they can't really use.

Reply:
Excellent point. I wasn't thinking super hard as I was just moving stuff around.

Reply:
Both are private now. Given that we have the FilterData API, I think the FilterOption should be private since its just an implementation detail. Your point on them showing up in the docs, yet unusable sold me.